src/dataSource.js

```diff
@@ -90,84 +87,201 @@ class DataSource {
   }
 
   /**
-   * Returns a single row of the data (array or object, depending on what you have). `row` is the index of the row in the data source.
+   * Returns a single row of the data or a subset of its columns. If a column range or `toArray` arguments are provided, it
+   * operates only on the columns declared by the `columns` setting or the data schema.
    *
    * @param {number} row Physical row index.
+   * @param {number} [startColumn] Starting index for the column range (optional).
+   * @param {number} [endColumn] Ending index for the column range (optional).
+   * @param {boolean} [toArray] `true` if the returned value should be forced to be presented as an array.
    * @returns {Array|object}
    */
-  getAtRow(row) {
-    return this.data[row];
+  getAtRow(row, startColumn, endColumn, toArray = false) {
+    const getAllProps = startColumn === void 0 && endColumn === void 0;
+    let dataRow = null;
+    let newDataRow = null;
+    let modifyRowData = null;
+
+    if (this.hot.hasHook('modifyRowData')) {
+      modifyRowData = this.hot.runHooks('modifyRowData', row);
+    }
+
+    dataRow = modifyRowData !== null && isNaN(modifyRowData) ? modifyRowData : this.data[row];
```

Comment:
We've got `isNaN` check here, `Number.isNumber` somewhere else. Why do we have to check that?

We have `isArray`, `isObject` and `isFunction` checks later. `null` can be a valid way to modify row data. The only case is `undefined` when there is nothing returned from the hook?